www/~/%username/emails/modify.json.spt

```diff
@@ -27,11 +27,18 @@ if not participant.email_lang:
 
 msg = None
 if action in ('add-email', 'resend'):
-    r = participant.add_email(address, website.env.resend_verification_threshold)
-    if r:
-        msg = _("A verification email has been sent to {0}.", address)
+    try:
+        participant.add_email(address, website.env.resend_verification_threshold)
+    except EmailTaken:
```

Comment:
We're testing this now (not sure exactly where, but I saw failures related to this and fixed them.)